### ISSUE TITLE
[Bug]: App Hot Reload not working

### ISSUE THREAD

**Issue by raymondmatthew** [Shopify/cli#2389]
### Please confirm that you have:

- [X] Searched [existing issues](.) to see if your issue is a duplicate. (If you’ve found a duplicate issue, feel free to add additional information in a comment on it.)
- [X] Reproduced the issue in [the latest CLI version](https://www.npmjs.com/package/@shopify/cli).

### In which of these areas are you experiencing a problem?

App

### Expected behavior

Saving /frontend/pages/index.jsx will update front end automatically or at least by refresh

### Actual behavior

need to restart app.

### Verbose output

```shell
2023-07-11 05:36:32 │ web-backend  │ npm
2023-07-11 05:36:32 │ web-backend  │  verb cli /home/dcoder/.nvm/versions/node/v20.4.0/bin/node /home/dcoder/.nvm/versions/node/v20.4.0/bin/npm
2023-07-11 05:36:32 │ web-backend  │ npm info using npm@9.8.0
2023-07-11 05:36:32 │ web-backend  │ npm info using node@v20.4.0
2023-07-11 05:36:32 │ web-backend  │ npm info found workspace root at /mnt/d/Work/forks/apps/test-order
2023-07-11 05:36:32 │ web-backend  │ npm verb title npm run dev
2023-07-11 05:36:32 │ web-backend  │ npm verb argv "run" "dev"
2023-07-11 05:36:32 │ web-backend  │ npm verb logfile logs-max:10 dir:/home/dcoder/.npm/_logs/2023-07-11T05_36_32_343Z-
2023-07-11 05:36:32 │ web-backend  │ npm verb logfile /home/dcoder/.npm/_logs/2023-07-11T05_36_32_343Z-debug-0.log
2023-07-11 05:36:32 │ web-frontend │ npm verb cli /home/dcoder/.nvm/versions/node/v20.4.0/bin/node /home/dcoder/.nvm/versions/node/v20.4.0/bin/npm
2023-07-11 05:36:32 │ web-frontend │ npm info using npm@9.8.0
2023-07-11 05:36:32 │ web-frontend │ npm info using node@v20.4.0
2023-07-11 05:36:32 │ web-frontend │ npm info found workspace root at /mnt/d/Work/forks/apps/test-order
2023-07-11 05:36:32 │ web-frontend │ npm verb title npm run dev
2023-07-11 05:36:32 │ web-frontend │ npm verb argv "run" "dev"
2023-07-11 05:36:32 │ web-frontend │ npm verb logfile logs-max:10 dir:/home/dcoder/.npm/_logs/2023-07-11T05_36_32_340Z-
2023-07-11 05:36:32 │ web-frontend │ npm verb logfile /home/dcoder/.npm/_logs/2023-07-11T05_36_32_340Z-debug-0.log
2023-07-11 05:36:32 │ web-frontend │
2023-07-11 05:36:32 │ web-frontend │ > shopify-frontend-template-react@1.0.0 dev
2023-07-11 05:36:32 │ web-frontend │ > vite
2023-07-11 05:36:32 │ web-frontend │
2023-07-11 05:36:32 │ web-backend  │
2023-07-11 05:36:32 │ web-backend  │ > shopify-app-template-node@1.0.0 dev
2023-07-11 05:36:32 │ web-backend  │ > cross-env NODE_ENV=development nodemon index.js --ignore ./frontend
2023-07-11 05:36:32 │ web-backend  │
2023-07-11 05:36:34 │ web-backend  │ [nodemon] 2.0.22
2023-07-11 05:36:34 │ web-backend  │ [nodemon] to restart at any time, enter `rs`
2023-07-11 05:36:34 │ web-backend  │ [nodemon] watching path(s): *.*
2023-07-11 05:36:34 │ web-backend  │ [nodemon] watching extensions: js,mjs,json
2023-07-11 05:36:34 │ web-backend  │ [nodemon] starting `node index.js`
2023-07-11 05:36:36 │ web-frontend │
2023-07-11 05:36:36 │ web-frontend │   VITE v4.4.2  ready in 4471 ms
2023-07-11 05:36:36 │ web-frontend │
2023-07-11 05:36:37 │ web-frontend │   ➜  Local:   http://localhost:40301/
2023-07-11 05:36:44 │ web-backend  │ [shopify-api/INFO] version 7.4.0, environment Node v20.4.0
```

### Reproduction steps

1. npm run dev
2. change /frontend/pages/index.jsx
3. save

### Operating System

Windows 11

### Shopify CLI version (check your project's `package.json` if you're not sure)

3.47.3

### Node version (run `node -v` if you're not sure)

v20.4.0

### What language and version are you using in your application?

node

**Comment by alvaro-shopify** [comment]
Hi @raymondmatthewm, thanks for rising up this issue!. I have just created a node app from scratch and hotcode reloading is working for me.
When you modify any file you should see a message similar to this one:
```
2023-07-13 14:36:19 │ backend    │ [shopify-app/INFO] Request session has a valid access token | {shop: cli-second.myshopify.com}
2023-07-13 14:36:40 │ frontend   │ 4:36:40 PM [vite] hmr update /pages/index.jsx
```

Make sure when you open the app in the browser that the `ws` connections used for the hot code reloading feature is open and working. The url should be different because it's a dynamic one created by cloudfare

![image](https://github.com/Shopify/cli/assets/105213827/1f81d520-9eea-4c82-8ab3-c267a9ecf791)


Please, could you confirm if you see any error?

**Comment by raymondmatthew** [comment]
Hello, my changes are reflected now.
My issue seems WSL2 related.  I recently updated from WSL1 to WSL2.

When using WSL2, it seems that it's not watching the folder files, when I moved my project to the Ubuntu file system it instantly worked.

However, if anyone there is using WSL2 and knows how to fix this, let me know.  It's quite an inconvenience to manage two filesystems and WSL can easily be uninstalled nowadays so there is a little risk with that.
